mbed test is not working in mbed-os-example-bootloader application

### Description
- Type: Bug 
- Priority: Minor

I imported mbed-os-example-bootloader application and tried to run mbed test with it, but it is not working. 

```
$ mbed test --list
Traceback (most recent call last):
  File "C:\ARM\SourceCode\mbed-os-example-bootloader\mbed-os\tools\test.py", line 128, in <module>
    app_config=options.app_config))
  File "C:\ARM\SourceCode\mbed-os-example-bootloader\mbed-os\tools\test_api.py", line 2017, in find_tests
    silent=True, app_config=app_config)
  File "C:\ARM\SourceCode\mbed-os-example-bootloader\mbed-os\tools\build_api.py", line 339, in prepare_toolchain
    add_regions_to_profile(build_profile, config, cur_tc)
  File "C:\ARM\SourceCode\mbed-os-example-bootloader\mbed-os\tools\build_api.py", line 290, in add_regions_to_profile
    profile["common"].append("-D%s=0x%x" %  define)
TypeError: 'NoneType' object has no attribute '__getitem__'
[ERROR] 'NoneType' object has no attribute '__getitem__'
[mbed] ERROR: "python" returned error code 1.
[mbed] ERROR: Command "python -u C:\ARM\SourceCode\mbed-os-example-bootloader\mbed-os\tools\test.py -t ARM -m K64F --source C:\ARM\SourceCode\mbed-os-example-bootloader --build C:\ARM\SourceCode\mbed-os-example-bootloader\BUILD\tests\K64F\ARM --test-spec C:\ARM\SourceCode\mbed-os-example-bootloader\BUILD\tests\K64F\ARM\test_spec.json --list" in "C:\ARM\SourceCode\mbed-os-example-bootloader"
```

I did not touch any files in the repo.


tamkam01@E109896 MINGW64 /c/ARM/SourceCode/mbed-os-example-bootloader (master)
$ git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working tree clean

tamkam01@E109896 MINGW64 /c/ARM/SourceCode/mbed-os-example-bootloader/mbed-os (mbed-os-5.4)
$ git status
On branch mbed-os-5.4
Your branch is up-to-date with 'origin/mbed-os-5.4'.
nothing to commit, working tree clean

mbed cli version
tamkam01@E109896 MINGW64 /c/ARM/SourceCode/mbed-os-example-bootloader (master)
$ mbed --version
1.0.0

mbed os is the latest version
tamkam01@E109896 MINGW64 /c/ARM/SourceCode/mbed-os-example-bootloader/mbed-os (mbed-os-5.4)
$ git pull
Already up-to-date.

tamkam01@E109896 MINGW64 /c/ARM/SourceCode/mbed-os-example-bootloader/mbed-os (mbed-os-5.4)
$ git describe
latest

On mbed os master branch the same issue is seen.

Strange thing is that another example application (mbed-os-example-blinky) with the same mbed os version is working fine

--- a/tools/build_api.py
+++ b/tools/build_api.py
@@ -283,6 +283,8 @@ def add_regions_to_profile(profile, config, toolchain_class):
     config - the configuration object that owns the region
     toolchain_class - the class of the toolchain being used
     """
+    if not profile:
+        return
     regions = list(config.regions)
     for region in regions:
         for define in [(region.name.upper() + "_ADDR", region.start),

